src/Builder/DatagridBuilder.php

```diff
@@ -91,9 +91,9 @@ public function fixFieldDescription(FieldDescriptionInterface $fieldDescription)
             ModelFilter::class === $fieldDescription->getType() && null === $fieldDescription->getOption('field_type')
             || EntityType::class === $fieldDescription->getOption('field_type')
         ) {
-            $fieldDescription->mergeOption('field_options', [
+            $fieldDescription->setOption('field_options', array_merge([
```

Comment:
If these offsets are strings, we could use `+` instead of `array_merge()`.

Reply:
Do you prefer $a + $b over array_merge($b, $a) or do we merge like this ?

Reply:
IMO, avoiding a function call is better for performance reasons.

Reply:
I think array_merge is clearer and more descriptive. 

Performance wise I've just found this: https://stackoverflow.com/questions/57192782/array-merge-vs-direct-array-injection-performance/57193666 which indicates that array_merge is faster.

Reply:
It's just a suggestion, not a blocker for the fix.
We could make performance improvements in a new PR.
For the records: https://gist.github.com/Ocramius/8399625.

Reply:
@phansys not sure it's still true, the comparison was done with PHP 5.4